Code/GraphMol/MolInterchange/Parser.cpp

```diff
@@ -0,0 +1,469 @@
+    const auto &miter = rjDefaults.FindMember(key);
+    if (miter != rjDefaults.MemberEnd()) {
+      if (!miter->value.IsBool())
+        throw FileParseException(std::string("Bad format: value of ") +
+                                 std::string(key) +
+                                 std::string(" is not a bool"));
+      bool res = miter->value.GetBool();
+      boolMap[key] = res;
+      return res;
+    }
+    return false;
+  }
+  std::string getString(const char *key) const {
+    PRECONDITION(key, "no key");
+    const auto &lookup = stringMap.find(key);
+    if (lookup != stringMap.end()) return lookup->second;
+    const auto &miter = rjDefaults.FindMember(key);
+    if (miter != rjDefaults.MemberEnd()) {
+      if (!miter->value.IsString())
+        throw FileParseException(std::string("Bad format: value of ") +
+                                 std::string(key) +
+                                 std::string(" is not a string"));
+      std::string res = miter->value.GetString();
+      stringMap[key] = res;
+      return res;
+    }
+    return "";
+  }
+};
+
+int getIntDefaultValue(const char *key, const rj::Value &from,
+                       const DefaultValueCache &defaults) {
+  PRECONDITION(key, "no key");
+  auto endp = from.MemberEnd();
+  auto miter = from.FindMember(key);
+  if (miter != endp) {
+    if (!miter->value.IsInt())
+      throw FileParseException(std::string("Bad format: value of ") +
+                               std::string(key) +
+                               std::string(" is not an int"));
+    return miter->value.GetInt();
+  }
+  return defaults.getInt(key);
+}
+bool getBoolDefaultValue(const char *key, const rj::Value &from,
+                         const DefaultValueCache &defaults) {
+  PRECONDITION(key, "no key");
+  auto endp = from.MemberEnd();
+  auto miter = from.FindMember(key);
+  if (miter != endp) {
+    if (!miter->value.IsBool())
+      throw FileParseException(std::string("Bad format: value of ") +
+                               std::string(key) +
+                               std::string(" is not a bool"));
+    return miter->value.GetBool();
+  }
+  return defaults.getBool(key);
+}
+std::string getStringDefaultValue(const char *key, const rj::Value &from,
+                                  const DefaultValueCache &defaults) {
+  PRECONDITION(key, "no key");
+  auto endp = from.MemberEnd();
+  auto miter = from.FindMember(key);
+  if (miter != endp) {
+    if (!miter->value.IsString())
+      throw FileParseException(std::string("Bad format: value of ") +
+                               std::string(key) +
+                               std::string(" is not a string"));
+    return miter->value.GetString();
+  }
+  return defaults.getString(key);
+}
+
+void readAtom(RWMol *mol, const rj::Value &atomVal,
+              const DefaultValueCache &atomDefaults) {
+  PRECONDITION(mol, "no mol");
+  Atom *at = new Atom(getIntDefaultValue("z", atomVal, atomDefaults));
```

Comment:
I would consider namespacing "z" and all other string constants.  This might also help serve for format documentation.